lib/workers/repository/reconfigure/index.ts

```diff
@@ -0,0 +1,158 @@
+
+  // only use valid cached information
+  if (reconfigureCache?.reconfigureBranchSha === branchSha) {
+    logger.debug('Cache is valid. Skipping validation check');
+    return;
+  }
+
+  try {
+    await scm.checkoutBranch(branchName);
+    configFileName = await detectConfigFile();
+  } catch (err) {
+    /*istanbul ignore next - should never happen*/
+    logger.error(
+      { err },
+      'Error while searching for config file in reconfigure branch'
+    );
+  }
+
+  if (!is.nonEmptyString(configFileName)) {
+    logger.warn('No config file found in reconfigure branch');
+    await platform.setBranchStatus({
+      branchName,
+      context,
+      description: 'Validation Failed - No config file found',
+      state: 'red',
+    });
+    setReconfigureBranchCache(branchSha!, false);
+    await scm.checkoutBranch(config.baseBranch!);
+    return;
+  }
+
+  let configFileRaw: string | null = null;
+  try {
+    configFileRaw = await readLocalFile(configFileName, 'utf8');
+  } catch (err) {
+    /*istanbul ignore next - should never happen*/
+    logger.error('Error while reading config file');
+  }
+
+  if (!is.nonEmptyString(configFileRaw)) {
+    logger.warn('Empty or invalid config file');
+    await platform.setBranchStatus({
+      branchName,
+      context,
+      description: 'Validation Failed - Empty/Invalid config file',
+      state: 'red',
+    });
+    setReconfigureBranchCache(branchSha!, false);
+    await scm.checkoutBranch(config.baseBranch!);
+    return;
+  }
+
+  let configFileParsed: any;
+  try {
+    const fileType = upath.extname(configFileName);
+    configFileParsed = JSON5.parse(configFileRaw);
+    if (fileType === '.json') {
+      // no need to confirm renovate field in package.json we already do it in `detectConfigFile()`
+      if (configFileName === 'package.json') {
+        configFileParsed = configFileParsed.renovate;
+      }
+    }
+  } catch (err) {
+    logger.error({ err }, 'Error while reading config file');
+    await scm.checkoutBranch(config.baseBranch!);
+    return;
+  }
+
+  // perform validation and provide a passing or failing check run based on result
+  const validationResult = await validateConfig(configFileParsed);
+
+  // failing check
+  if (validationResult.errors.length > 0) {
+    // add comment to reconfigure PR if it exists
+    const branchPr = await platform.getBranchPr(branchName, config.baseBranch);
+    if (branchPr) {
+      let body = `There is an error with this repository's Renovate configuration that needs to be fixed.\n\n`;
+      body += `Location: \`${configFileName}\`\n`;
+      body += `Message: \`${validationResult.errors
+        .map((e) => e.message)
+        .join(', ')
+        .replace(regEx(/`/g), "'")}\`\n`;
+
+      await ensureComment({
+        number: branchPr.number,
+        topic: 'Action Required: Fix Renovate Configuration',
+        content: body,
+      });
+    }
+    // log the errors
+    logger.debug(
```

Comment:
Move this to the top of the if clause. 
We want to first log and then handle the PR else if something fails during the PR update we will have no log entry either.